src/com/company/exercises/CombinedExerciseStreams.java

```diff
@@ -0,0 +1,98 @@
+package com.company.exercises;
+
+
+import com.company.data.MusicSampleData;
+import com.company.models.Album;
+import com.company.models.Artist;
+import com.company.models.Track;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CombinedExerciseStreams {
+//    Our first problem to solve is, for a given album, to find the nationality of every band playing on that album.
```

Comment:
The formatting of your comments can be improved in general.
There should be a good reason to leave a comment in your code. Permanent comments should have the same indentation as your code and should have a single empty space right after the double slash.

Reply:
Done